pkg/virt-handler/vm.go

```diff
@@ -1904,18 +1904,38 @@ func (d *VirtualMachineController) isPreMigrationTarget(vmi *v1.VirtualMachineIn
 }
 
 func (d *VirtualMachineController) checkNetworkInterfacesForMigration(vmi *v1.VirtualMachineInstance) error {
-	networks := map[string]*v1.Network{}
-	for _, network := range vmi.Spec.Networks {
-		networks[network.Name] = network.DeepCopy()
+	err := validatePodNetworkInterfaceUsesMasqueradeBinding(vmi)
+	if err != nil {
+		return err
 	}
+
+	return nil
+}
+
+func validatePodNetworkInterfaceUsesMasqueradeBinding(vmi *v1.VirtualMachineInstance) error {
+	interfacesByName := map[string]v1.Interface{}
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Masquerade == nil && networks[iface.Name].Pod != nil {
-			return fmt.Errorf("cannot migrate VMI which does not use masquerade to connect to the pod network")
-		}
+		interfacesByName[iface.Name] = iface
 	}
+
+	vmiPodNetworkName := lookupVMIPodNetworkName(vmi.Spec.Networks)
+	if vmiPodNetworkName != "" && interfacesByName[vmiPodNetworkName].Masquerade == nil {
```

Comment:
Self-note: I think the previous code also assumed that for each network there is an interface. So we should be safe here.

Reply:
It is validated by the admmiter code, if you create an interface it must be associated with a network
This is also correct for the counterpart if you have a network it must be associated an interface.